test/pylib/util.py

```diff
@@ -77,4 +79,24 @@ async def get_hosts() -> Optional[list[Host]]:
     return hosts
 
 
+async def get_available_host(cql: Session, deadline: float) -> Host:
+    hosts = cql.cluster.metadata.all_hosts()
+    async def find_host():
+        for h in hosts:
+            try:
+                await cql.run_async(
+                    "select key from system.local where key = 'local'", host=h)
+            except NoHostAvailable:
+                logging.debug(f"get_available_host: {h} not available")
+                continue
+            return h
+        return None
+    return await wait_for(find_host, deadline)
+
+
+async def read_barrier(cql: Session, host: Host):
+    with pytest.raises(InvalidRequest, match="nosuch"):
```

Comment:
It would be nice to have a docstring or comment explaining why this works.

From the original commit:

https://github.com/scylladb/scylladb/commit/e7c9ca560bf292ccdb2aba03fd04367efff1cee9

```
Raft replication doesn't guarantee that all replicas see
identical Raft state at all times, it only guarantees the
same order of events on all replicas.

When comparing raft state with gossip state on a node, first
issue a read barrier to ensure the node has the latest raft state.

To issue a read barrier it is sufficient to alter a non-existing
state: in order to validate the DDL the node needs to sync with the
leader and fetch its latest group0 state.
```